socket_linux.go

```diff
@@ -26,13 +31,26 @@ func NewSocket(addr *net.UDPAddr, recvBuf int, reuseport bool) (net.PacketConn,
 		return nil, err
 	}
 
-	sockaddr := unix.SockaddrInet4{
-		Port: addr.Port,
-	}
-	if copied := copy(sockaddr.Addr[:], addr.IP); copied != net.IPv4len {
-		panic("did not copy enough bytes of ip address")
+	var sa unix.Sockaddr
+	if domain == unix.AF_INET {
+		sockaddr := &unix.SockaddrInet4{
+			Port: addr.Port,
+		}
+		if copied := copy(sockaddr.Addr[:], addr.IP); copied != net.IPv4len {
+			panic("did not copy enough bytes of ip address")
+		}
+		sa = sockaddr
+	} else {
+		sockaddr := &unix.SockaddrInet6{
+			Port: addr.Port,
+		}
+		// addr.IP will be length 0 for "bind all interfaces"
+		if copied := copy(sockaddr.Addr[:], addr.IP); !(copied == net.IPv6len || copied == 0) {
```

Comment:
Yes, that will match. Same thing though: I still need to use copy, but could possibly omit the paranoid error checking